sql-statements/sql-statement-foreign-key.md

```diff
@@ -0,0 +1,255 @@
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+```
+
+## Foreign key constraint check
+
+Whether to enable the foreign key constraint check is controlled by the system variable [`foreign_key_checks`](/system-variables.md#foreign_key_checks). By default, this variable value is set to `ON`, meaning that the foreign key constraint check is enabled. This variable has two scopes: `GLOBAL` and `SESSION`. Keeping this variable enabled can ensure the integrity of foreign key reference relationships.
+
+The effect of disabling [`foreign_key_checks`](/system-variables.md#foreign_key_checks) is as follows:
+
+- When deleting a parent table referenced by a foreign key, the deletion can succeed only when the foreign key constraint check is disabled.
+- When importing data to a database, the order of creating tables might be different from the foreign key dependency order, which might cause the creation of tables to fail. Only when the foreign key constraint check is disabled can the tables be created successfully. In addition, when importing data, disabling the foreign key constraint check can also speed up data import.
+- When importing data to a database, if the data of the child table is imported first, an error will be reported. Only when the foreign key constraint check is disabled can the data of the child table be imported successfully.
+- When executing `ALTER TABLE` operations related to foreign keys, the foreign key constraint check must be enabled to execute successfully.
+
+When the foreign key constraint check is disabled, the foreign key constraint check and reference operation are not executed, except for the following scenarios:
+
+- When executing `ALTER TABLE`, if the definition of the foreign key is incorrect, an error is still reported.
+- When deleting the index required by the foreign key, the foreign key must be deleted first. Otherwise, an error is reported when deleting the foreign key.
+- When creating a foreign key, if it does not meet the conditions or restrictions of the foreign key, an error is reported.
+
+## Lock
+
+When `INSERT` or `UPDATE` a child table, the foreign key constraint checks whether the corresponding foreign key value exists in the parent table, and locks the row in the parent table to avoid the foreign key value being deleted by other operations violating the foreign key constraint. The locking behavior is equivalent to performing a `SELECT FOR UPDATE` operation on the row where the foreign key value is located in the parent table. Because TiDB currently does not support `LOCK IN SHARE MODE`, if there are a lot of concurrent writes to the child table and most of the referenced foreign key values are the same, there might be serious lock conflicts. It is recommended to disable [`foreign_key_checks`](/system-variables.md#foreign_key_checks) when writing a large number of child table data.
+
+## Definition and metadata of foreign keys
```

Comment:
Looks like `information_schema.table_constraints` should also be mentioned.

```
sql> SELECT * FROM information_schema.table_constraints WHERE CONSTRAINT_TYPE='FOREIGN KEY'\G
*************************** 1. row ***************************
CONSTRAINT_CATALOG: def
 CONSTRAINT_SCHEMA: test
   CONSTRAINT_NAME: fk_1
      TABLE_SCHEMA: test
        TABLE_NAME: t2
   CONSTRAINT_TYPE: FOREIGN KEY
1 row in set (0.0025 sec)
```

And `information_schema.referential_constraints`:
```
sql> TABLE information_schema.referential_constraints\G
*************************** 1. row ***************************
       CONSTRAINT_CATALOG: def
        CONSTRAINT_SCHEMA: test
          CONSTRAINT_NAME: fk_1
UNIQUE_CONSTRAINT_CATALOG: def
 UNIQUE_CONSTRAINT_SCHEMA: test
   UNIQUE_CONSTRAINT_NAME: PRIMARY
             MATCH_OPTION: NONE
              UPDATE_RULE: NO ACTION
              DELETE_RULE: CASCADE
               TABLE_NAME: t2
    REFERENCED_TABLE_NAME: t1
1 row in set (0.0021 sec)
```

Reply:
Great, Nice catch, I will add those in https://github.com/pingcap/docs-cn/pull/11974 first.